core/server/master/src/main/java/alluxio/master/meta/UpdateCheck.java

```diff
@@ -98,7 +111,46 @@ public static String getUserAgentString(String clusterID) throws IOException {
         sysInfo = joiner.join(sysInfo, joiner.join(ec2Info));
       }
     }
-    return String.format("Alluxio/%s (%s)", ProjectConstants.VERSION, sysInfo);
+    return sysInfo;
+  }
+
+  /**
+   * Get the features information.
+   *
+   * @return a list of string represent feature names
+   */
+  @VisibleForTesting
+  public static String getFeatureString() {
+    Joiner joiner = Joiner.on("; ").skipNulls();
+    List<String> featureInfo = new ArrayList<>();
+    if (FeatureUtils.isEmbeddedJournal()) {
+      featureInfo.add("embedded");
+    }
+    if (FeatureUtils.isRocks()) {
+      featureInfo.add("rocks");
+    }
+    if (FeatureUtils.isZookeeperEnable()) {
+      featureInfo.add("zk");
+    }
+    if (FeatureUtils.isBackupDelegationEnable()) {
+      featureInfo.add("backupDelegation");
+    }
+    if (FeatureUtils.isDailyBackupEnable()) {
+      featureInfo.add("dailyBackup");
+    }
+    if (!FeatureUtils.isPersistenceBlacklistEmpty()) {
+      featureInfo.add("persistBlackList");
+    }
+    if (FeatureUtils.isUnsafeDirectPersistEnable()) {
+      featureInfo.add("unsafePersist");
+    }
+    if (FeatureUtils.isMasterAuditLoggingEnable()) {
+      featureInfo.add("masterAuditLog");
+    }
+    if (featureInfo.size() == 0) {
+      return null;
```

Comment:
should the method be annotated somehow if this method can return null?
is there a unit test for this corner case?